### PR TITLE
Lanzatool: fix config limit

### LINKED ISSUE #46: Bootloader configuration deleted
As we can see in https://bin.hubrecht.ovh/?b376f2a3f8e744c3#8hdQHWaR7zvZCtYikanp1CdTL16z5jgD6UBV6b4SECW4 , when we run `nixos-rebuild switch` with lanzaboote activated, it deletes almost all directories in `/boot/EFI/, but the directory `/boot/EFI/loader` is used to store the configuration of the bootloader, so it is erased at each rebuild. This is very annoying as when we use systemd-boot there is no longer a timeout and we cannot choose which generation we want to boot on.

### PR DESCRIPTION
**Pull request by nikstur**
This should fix #46. 
Depends on #48.

### CODE REVIEW

File: rust/lanzatool/src/gc.rs [hunk 1]
```diff
@@ -35,10 +35,30 @@ impl Roots {
...
+        mut predicate: P,
+    ) -> Result<()>
+    where
+        P: FnMut(&Path) -> bool,
+    {
         // Find all the paths not used anymore.
         let entries_not_in_use = WalkDir::new(directory.as_ref())
             .into_iter()
-            .filter(|e| !self.in_use(e.as_ref().ok()));
+            .filter(|e| !self.in_use(e.as_ref().ok()))
```
Comment (RaitoBezarius) [inline rust/lanzatool/src/gc.rs:57]: I sure hope that Rust performs stream fusion

**Review by RaitoBezarius — APPROVED** [review]
Great! I will sleep at night now!

File: rust/lanzatool/src/gc.rs (hunk 1 above)
Comment (nikstur) [reply rust/lanzatool/src/gc.rs:57]: No idea, but I trust rustc will do the right thing™

Comment (RaitoBezarius) [reply rust/lanzatool/src/gc.rs:57]: inchallah as we say

**Pull request merged**